lottie/src/main/java/com/airbnb/lottie/utils/OffscreenLayer.java

```diff
@@ -0,0 +1,544 @@
+
+    if (bounds.width() >= bitmap.getWidth() || bounds.height() >= bitmap.getHeight()) {
+      return true;
+    }
+
+    // If the required area has reduced in size considerably, trigger a reallocation, since
+    // we might be paying a large unnecessary penalty to work with a bitmap that big.
+    return bounds.width() < bitmap.getWidth() * 0.75f || bounds.height() < bitmap.getHeight() * 0.75f;
+  }
+
+  public Canvas start(Canvas parentCanvas, RectF bounds, ComposeOp op) {
+    if (this.parentCanvas != null) {
+      throw new IllegalStateException("Cannot nest start() calls on a single OffscreenBitmap - call finish() first");
+    }
+
+    // Determine the scaling applied by the parentCanvas' pre-existing transform matrix. This is an optimization
+    // to avoid creating bitmaps (or render nodes) with unreasonable sizes that will get scaled down when drawn
+    // onto parentCanvas anyhow.
+    if (preExistingTransform == null) preExistingTransform = new float[9];
+    if (parentCanvasMatrix == null) parentCanvasMatrix = new Matrix();
+    parentCanvas.getMatrix(parentCanvasMatrix);
+    parentCanvasMatrix.getValues(preExistingTransform);
+
+    float pixelScaleX = preExistingTransform[Matrix.MSCALE_X];
+    float pixelScaleY = preExistingTransform[Matrix.MSCALE_Y];
+
+    if (scaledBounds == null) scaledBounds = new RectF();
+    scaledBounds.set(
+        bounds.left * pixelScaleX,
+        bounds.top * pixelScaleY,
+        bounds.right * pixelScaleX,
+        bounds.bottom * pixelScaleY
+    );
+
+    this.parentCanvas = parentCanvas;
+    this.op = op;
+    this.currentStrategy = chooseRenderStrategy(parentCanvas, op);
+    if (this.targetRect == null) this.targetRect = new Rect();
+    this.targetRect.set((int)bounds.left, (int)bounds.top, (int)bounds.right, (int)bounds.bottom);
+
+    if (composePaint == null) composePaint = new LPaint();
+    composePaint.reset();
+
+    Canvas childCanvas;
+    switch (currentStrategy) {
+      case DIRECT:
+        childCanvas = parentCanvas;
+        childCanvas.save();
+        break;
+
+      case SAVE_LAYER:
+        // Paint to use for composition
+        composePaint.setAlpha(op.alpha);
+        composePaint.setColorFilter(op.colorFilter);
+        if (op.hasBlendMode()) {
+          PaintCompat.setBlendMode(composePaint, op.blendMode);
+        }
+
+        // This adds an entry in the parentCanvas stack, will be popped and composited
+        // when restore() is called
+        Utils.saveLayerCompat(parentCanvas, bounds, composePaint);
+        childCanvas = parentCanvas;
+        break;
+
+      case BITMAP:
+        if (clearPaint == null) {
+          clearPaint = new LPaint();
+          clearPaint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.CLEAR));
+        }
+
+        if (needNewBitmap(bitmap, scaledBounds)) {
+          if (bitmap != null) {
+            deallocateBitmap(bitmap);
+          }
+          bitmap = allocateBitmap(scaledBounds, Bitmap.Config.ARGB_8888);
```

Comment:
```suggestion
        if (needNewBitmap(bitmap, scaledBounds)) {
          if (bitmap != null) {
            deallocateBitmap(bitmap);
          }
          bitmap = allocateBitmap(scaledBounds, Bitmap.Config.ARGB_8888);
```

I think it would be great if you could encapsulate the `needNew` + `deallocate` + `allocate` into a single helper that returns the bitmap.

Then, in each of the functions that calls the new function (something like `getBitmap(Rect)`, it would just use the return value as a local variable rather than using fields on the object